app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/derivepublickey/DerivePublicKeyViewModel.kt

```diff
@@ -130,16 +148,12 @@ class DerivePublicKeyViewModel @Inject constructor(
             is FactorSource.Device -> {
                 derivePublicKeyFromDeviceFactorSource(
                     forNetworkId = input.forNetworkId,
-                    deviceFactorSource = factorSource
+                    deviceFactorSource = factorSource,
+                    entityKind = input.entityKind
                 )
             }
 
             is FactorSource.Ledger -> {
-                _state.update { uiState ->
-                    uiState.copy(
-                        showContentForFactorSource = ShowContentForFactorSource.Ledger(selectedLedgerDevice = factorSource)
-                    )
-                }
```

Comment:
I think I had it here and not in the init because the method `derivePublicKey()` can be called again when retry is clicked.
Is it working now? I mean after clicking retry?

Reply:
hmm actually it should be working fine, right?